common/authentication/postgresql/metadata.go

```diff
@@ -43,27 +44,34 @@ func (m *PostgresAuthMetadata) Reset() {
 	m.ConnectionMaxIdleTime = 0
 	m.MaxConns = 0
 	m.UseAzureAD = false
+	m.UseAWSIAM = false
 	m.QueryExecMode = ""
 }
 
+type InitWithMetadataOpts struct {
+	AzureADEnabled bool
+	AWSIAMEnabled  bool
+}
+
 // InitWithMetadata inits the object with metadata from the user.
 // Set azureADEnabled to true if the component can support authentication with Azure AD.
 // This is different from the "useAzureAD" property from the user, which is provided by the user and instructs the component to authenticate using Azure AD.
-func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, azureADEnabled bool) (err error) {
+func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts InitWithMetadataOpts) (err error) {
 	// Validate input
-	if m.ConnectionString == "" {
-		return errors.New("missing connection string")
-	}
-
-	// Populate the Azure environment if using Azure AD
-	if azureADEnabled && m.UseAzureAD {
+	switch {
+	case opts.AzureADEnabled && m.UseAzureAD:
 		m.azureEnv, err = azure.NewEnvironmentSettings(meta)
 		if err != nil {
 			return err
 		}
-	} else {
-		// Make sure this is false
+	case opts.AWSIAMEnabled && m.UseAWSIAM:
```

Comment:
aws region should be a required metadata if useAWSIAM is set, no?

Reply:
yes, this is marked as required in the metadata yaml file